File: apps/macos-client/Sources/Models/Comment.swift
import Foundation

struct Comment: Codable, Identifiable {
    let id: UUID
    let contentId: UUID
    let authorId: UUID
    let parentId: UUID?
    let body: String
    let createdAt: Date
    let updatedAt: Date
    let author: UserPublic
    let replyCount: Int

    enum CodingKeys: String, CodingKey {
        case id, body, author
        case contentId = "content_id"
        case authorId = "author_id"
        case parentId = "parent_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case replyCount = "reply_count"
    }
}

struct CommentCreate: Encodable {
    let body: String
    let parentId: UUID?

    enum CodingKeys: String, CodingKey {
        case body
        case parentId = "parent_id"
    }
}

File: services/api/app/schemas/comment.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.schemas.user import UserPublic


class CommentBase(BaseModel):
    body: str


class CommentCreate(CommentBase):
    parent_id: UUID | None = None


class CommentUpdate(BaseModel):
    body: str


class Comment(CommentBase):
    id: UUID
    content_id: UUID
    author_id: UUID
    parent_id: UUID | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CommentWithAuthor(Comment):
    author: UserPublic
    reply_count: int = 0
